src/manager.cpp

```diff
@@ -110,9 +110,12 @@ Manager::Manager(
                                [this]() { this->performVPDRecollection(); });
 
         // Indicates FRU VPD collection for the system has not started.
-        iFace->register_property(
-            "CollectionStatus", std::string("NotStarted"),
-            sdbusplus::asio::PropertyPermission::readWrite);
+        iFace->register_property_rw<std::string>(
+            "CollectionStatus", sdbusplus::vtable::property_::emits_change,
+            [this](const std::string l_currStatus, const auto&) {
+            m_vpdCollectionStatus = l_currStatus;
+            return 0;
```

Comment:
Where are we setting the "NotStarted" value now?

Reply:
that is the default state, which is assigned to the variable m_vpdCollectionStatus

Reply:
ok, but on Dbus, to the property how the value will be assigned?

Reply:
the lambda at line 118 is the getter function. so when you do get-property the former is called and returns the current status

Reply:
here is the test done for this change. I didn't find any issue.

root@valid-hostname:/tmp# systemctl restart com.ibm.VPD.Manager
root@valid-hostname:/tmp# busctl get-property com.ibm.VPD.Manager /com/ibm/VPD/Manager com.ibm.VPD.Manager    CollectionStatus
s "NotStarted"
root@valid-hostname:/tmp# busctl get-property com.ibm.VPD.Manager /com/ibm/VPD/Manager com.ibm.VPD.Manager    CollectionStatus
s "InProgress"
root@valid-hostname:/tmp# busctl get-property com.ibm.VPD.Manager /com/ibm/VPD/Manager com.ibm.VPD.Manager    CollectionStatus
s "InProgress"
root@valid-hostname:/tmp# busctl get-property com.ibm.VPD.Manager /com/ibm/VPD/Manager com.ibm.VPD.Manager    CollectionStatus
s "Completed"
root@valid-hostname:/tmp#

Reply:
Ok, got it, I was just thinking from where the default value will be set on DBus but the return value below will do that. Cool.